lib/block-supports/duotone.php

```diff
@@ -0,0 +1,127 @@
+			);
+		}
+	}
+}
+
+
+/**
+ * Add CSS classes and inline styles for colors to the incoming attributes array.
+ * This will be applied to the block markup in the front-end.
+ *
+ * @param  WP_Block_Type $block_type       Block type.
+ * @param  array         $block_attributes Block attributes.
+ *
+ * @return array Colors CSS classes and inline styles.
+ */
+function gutenberg_apply_duotone_support( $block_type, $block_attributes ) {
+	$attributes          = array();
+	$has_duotone_support = false;
+	if ( property_exists( $block_type, 'supports' ) ) {
+		$has_duotone_support = gutenberg_experimental_get( $block_type->supports, array( 'duotone' ), false );
+	}
+	if ( $has_duotone_support ) {
+		$has_duotone_attribute = array_key_exists( 'duotone', $block_attributes );
+
+		if ( $has_duotone_attribute ) {
+			$attributes['class'] = $block_attributes['duotone']['id'];
+		}
+	}
+
+	return $attributes;
+}
+
+/**
+ * Render out the duotone stylesheet and SVG.
+ *
+ * @param  WP_Block_Type $block_type       Block type.
+ * @param  array         $block_attributes Block attributes.
+ * @param  string        $block_content    Rendered block content.
+ *
+ * @return string filtered block content.
+ */
+function gutenberg_render_duotone_support( $block_type, $block_attributes, $block_content ) {
+	$duotone_support = false;
+	if ( property_exists( $block_type, 'supports' ) ) {
+		$duotone_support = gutenberg_experimental_get( $block_type->supports, array( 'duotone' ), false );
+	}
+
+	$has_duotone_attribute = array_key_exists( 'duotone', $block_attributes );
+
+	if (
+		! $duotone_support ||
+		! $has_duotone_attribute
+	) {
+		return $block_content;
+	}
+
+	$duotone_id     = $block_attributes['duotone']['id'];
+	$duotone_values = $block_attributes['duotone']['values'];
+
+	// Adding the block class as to not affect other blocks.
+	$block_class = gutenberg_get_block_default_classname( $block_type->name );
+	$scope       = '.' . $block_class . '.' . $duotone_id;
+
+	// Object | boolean | string | string[] -> boolean | string | string[].
+	$selector =
+		! is_array( $duotone_support ) || ! array_key_exists( 'edit', $duotone_support )
+			? $duotone_support
+			: $duotone_support['edit'];
+
+	// boolean | string | string[] -> boolean[] | string[].
+	$selector = is_array( $selector )
+		? $selector
+		: array( $selector );
+
+	// boolean[] | string[] -> string[].
+	$selector = array_map(
+		function ( $selector ) use ( $scope ) {
+			return is_string( $selector )
+				? $scope . ' ' . $selector
+				: $scope;
+		},
+		$selector
+	);
+
+	// string[] -> string.
+	$selector = implode( ', ', $selector );
+
+	$duotone = gutenberg_render_duotone_filter( $selector, $duotone_id, $duotone_values );
+
+	return $block_content . $duotone;
+}
+
+// Register the block support.
+WP_Block_Supports::get_instance()->register(
+	'duotone',
+	array(
+		'register_attribute' => 'gutenberg_register_duotone_support',
+		'apply'              => 'gutenberg_apply_duotone_support',
```

Comment:
This apply function only works if the  block is dynamic and uses `get_block_wrapper_attributes` function if I'm not wrong. How does this support static blocks?

cc @mcsf

Reply:
Right. Block Supports are, as of #26111 + #26192, explicitly up to the block author to consume at `render_block`. AFAIK we haven't rolled out any automations for static blocks.

In addition, I don't believe that the `'render_block' => 'gutenberg_render_duotone_support'` property passed in this `register` call is being used anywhere. I double-checked, in case something had changed recently. Am I missing something?

Reply:
@mcsf The filter is being called at the end of [lib/class-wp-block-supports.php](https://github.com/WordPress/gutenberg/pull/26752/files/533b01a0ce2fe5999a2de1c57be2b5c4dae05203#diff-db303ad0b7ec45abb3b605e5ad68b640d8262e776ef2a986e2ae5b35a2c724b0R279). I'm doing something similar to how `register_attribute` is called.

Reply:
For now, I switched this over to using the `render_block` hook directly. We can continue discussion on https://github.com/WordPress/wordpress-develop/pull/984 if it's something that's still worth adding later.